assets/model_monitoring/components/src/action_analyzer/action_detector_component/run.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Entry script for Action Detector."""
+
+import json
+import argparse
+import pandas
+from typing import List
+from action_analyzer.contracts.action_detectors.low_retrieval_score_index_action_detector import (
+    LowRetrievalScoreIndexActionDetector
+)
+from action_analyzer.contracts.llm_client import LLMClient
+from action_analyzer.contracts.utils.detector_utils import (
+    get_index_id_from_index_content,
+    deduplicate_actions,
+    write_actions,
+    add_action_tag_to_run
+)
+from shared_utilities.constants import (
+    RETRIEVAL_SPAN_TYPE,
+    ROOT_SPAN_COLUMN,
+    GSQ_METRICS_LIST,
+    INDEX_ID_COLUMN
+)
+from shared_utilities.store_url import StoreUrl
+from shared_utilities.io_utils import try_read_mltable_in_spark
+from shared_utilities.span_tree_utils import SpanTree
+
+
+def parse_index_id(root_span: str) -> List[str]:
+    """Parse the span tree to get index id.
+
+    Args:
+        root_span(str): the span tree in json string format.
+
+    Returns:
+        list(str): list of parsed indexes in the span tree.
+    """
+    try:
+        tree = SpanTree.create_tree_from_json_string(root_span)
+        index_list = []
+        for span in tree:
+            if span.span_type == RETRIEVAL_SPAN_TYPE:
+                parent_id = span.parent_id
+                if not parent_id:
+                    print("No look up span found, skip action analyzer.")
+                    return None
+                index_span = tree.get_span_tree_node_by_span_id(parent_id)
+                print(index_span)
+                index_input = json.loads(index_span.input)
+                print(index_input)
+                index_content = index_input['mlindex_content']
+                index_list.append(get_index_id_from_index_content(index_content))
```

Comment:
get_index_id_from_index_content can return None, do we expect to have None in the list?

Reply:
added none check